services/repository/src/main/java/org/sagebionetworks/repo/web/service/VerificationServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+import org.sagebionetworks.repo.manager.NotificationManager;
+import org.sagebionetworks.repo.manager.UserManager;
+import org.sagebionetworks.repo.manager.VerificationManager;
+import org.sagebionetworks.repo.model.UserInfo;
+import org.sagebionetworks.repo.model.verification.VerificationPagedResults;
+import org.sagebionetworks.repo.model.verification.VerificationState;
+import org.sagebionetworks.repo.model.verification.VerificationStateEnum;
+import org.sagebionetworks.repo.model.verification.VerificationSubmission;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class VerificationServiceImpl implements VerificationService {
+	
+	@Autowired
+	private VerificationManager verificationManager;
+
+	@Autowired
+	private UserManager userManager;
+	
+	@Autowired
+	private NotificationManager notificationManager;
+
+	public VerificationServiceImpl() {}
+
+	// for testing
+	public VerificationServiceImpl(
+			VerificationManager verificationManager, 
+			UserManager userManager,
+			NotificationManager notificationManager) {
+		this.verificationManager=verificationManager;
+		this.userManager = userManager;
+		this.notificationManager = notificationManager;
+	}
+
+	@Override
+	public VerificationSubmission createVerificationSubmission(Long userId,
+			VerificationSubmission verificationSubmission,
+			String notificationUnsubscribeEndpoint) {
+		UserInfo userInfo = userManager.getUserInfo(userId);
+		VerificationSubmission result = verificationManager.
+				createVerificationSubmission(userInfo, verificationSubmission);
+		List<MessageToUserAndBody> createNotifications = verificationManager.
+				createSubmissionNotification(result, notificationUnsubscribeEndpoint);
+		notificationManager.sendNotifications(userInfo, createNotifications);
+		return result;
+	}
+	
+	public void deleteVerificationSubmission(Long userId, Long verificationId) {
+		UserInfo userInfo = userManager.getUserInfo(userId);
+		verificationManager.deleteVerificationSubmission(userInfo, verificationId);
+	}
+
+	@Override
+	public VerificationPagedResults listVerificationSubmissions(Long userId,
+			List<VerificationStateEnum> currentVerificationState,
+			Long verifiedUserId, long limit, long offset) {
+		UserInfo userInfo = userManager.getUserInfo(userId);
+		return verificationManager.listVerificationSubmissions(userInfo, currentVerificationState, verifiedUserId, limit, offset);
+	}
+
+	@Override
+	public void changeSubmissionState(Long userId,
+			long verificationSubmissionId, VerificationState newState,
+			String notificationUnsubscribeEndpoint) {
+		UserInfo userInfo = userManager.getUserInfo(userId);
+		verificationManager.changeSubmissionState(userInfo, verificationSubmissionId, newState);
+		List<MessageToUserAndBody> createNotifications = verificationManager.createStateChangeNotification(verificationSubmissionId, newState, notificationUnsubscribeEndpoint);
+		notificationManager.sendNotifications(userInfo, createNotifications);
```

Comment:
notification should always go through the asynchronous worker to leverage the workers' ability to retry when an error occurs.